### ISSUE TITLE
thrd_current() is broken on Windows

### ISSUE THREAD

**Issue by adam-frisby** [tinycthread/tinycthread#64]
thrd_current() calls [GetCurrentThread()](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentthread) which does not do what you would expect it to do.

GetCurrentThread() returns a "pseudo handle" for the calling thread. Which is kind of a substitute handle that always refers to the current thread. On x64 windows this value appears to be 0xfffffffffffffffe. Any call to GetCurrentThread() from any thread returns 0xfffffffffffffffe because that value always refers to the "current thread".

I was attempting to implement a shared mutex in which I need to keep track of what threads have already acquired the mutex. It was a surprise to me when I found that all my threads shared the exact same id.

The id returned by pthread_self() on the other hand is unique (at any given time, ids may be reused) and can be used to track threads.

[GetCurrentThreadId()](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentthreadid) on the other hand does return a unique thread ID, but returns a DWORD but thrd_t is defined as a HANDLE (which is not unique, multiple handles may refer to the same thread).

https://devblogs.microsoft.com/oldnewthing/20141015-00/?p=43843

**Comment by gyrovorbis** [comment]
This is totally true. I was bitten by this with a very similar scenario... Had to just restructure the code, because I couldn't figure an easy way around it...

...on the other hand, just so you know, they did just add initial support for C11 threads in the latest MSVC version: https://devblogs.microsoft.com/cppblog/c11-threads-in-visual-studio-2022-version-17-8-preview-2/
